python/paddle/distributed/fleet/meta_optimizers/sharding/utils.py

```diff
@@ -402,13 +402,16 @@ def get_grad_device(grad_name, shard):
     return shard.global_param2device[base_name]
 
 
-def get_first_check_finite_and_unscale_op_idx(block):
+def get_first_check_finite_and_unscale_op_idx(block, raise_error=True):
 
     for idx, op in enumerate(block.ops):
         if op.type == "check_finite_and_unscale":
             return idx
 
-    raise ValueError("check_finite_and_unscale does not exist in block")
+    if raise_error:
+        raise ValueError("check_finite_and_unscale does not exist in block")
```

Comment:
the error message should be:
"amp is turn on but check_finite_and_unscale op does not exist in main block"

Reply:
ok